src/main/java/org/folio/auth/authtokenmodule/tokens/Token.java

```diff
@@ -0,0 +1,262 @@
+      || claims.getString("sub").startsWith(Token.UNDEFINED_USER_NAME)) {
+        usesDummyPermissionsSource = true;
+        return true;
+      }
+
+    return false;
+  }
+
+  /**
+   * Will return true if this this token requires a check that the user is active.
+   * @param requestHeaders The headers in the http context where the token is being provided.
+   * @return True if the check should be made.
+   */
+  public boolean shouldCheckIfUserIsActive(MultiMap requestHeaders) {
+    var userId = requestHeaders.get(XOkapiHeaders.USER_ID);
+    return !usesDummyPermissionsSource && userId != null && !userId.trim().isEmpty();
+  }
+
+  /**
+   * Will add the calling module if the request headers require it.
+   * @param requestHeaders The headers in the http context where the token is being provided.
+   */
+  public void tryAddCallingModule(MultiMap requestHeaders) {
+    if (requestHeaders.contains(CALLING_MODULE_HEADER)) {
+      claims.put("calling_module", requestHeaders.get(CALLING_MODULE_HEADER));
+    }
+  }
+
+  /**
+   * Encodes the token as a JWT token.
+   * @param tokenCreator The TokenCreator to use to encode the token.
+   * @return The encoded token.
+   * @throws JOSEException
+   * @throws ParseException
+   */
+  public String encodeAsJWT(TokenCreator tokenCreator) throws JOSEException, ParseException {
+    String encodedClaims = claims.encode();
+    return tokenCreator.createJWTToken(encodedClaims);
+  }
+
+  /**
+   * Encodes the token as a JWE token.
+   * @param tokenCreator The TokenCreator to use to encode the token.
+   * @return The encoded token.
+   * @throws JOSEException
+   * @throws ParseException
+   */
+  public String encodeAsJWE(TokenCreator tokenCreator) throws JOSEException { 
+    String encodedClaims = claims.encode();
+    return tokenCreator.createJWEToken(encodedClaims);
+  }
+
+  /**
+   * Gets the claims from the provided JWT.
+   * @param jwt A string representing the JWT.
+   * @return A JsonObject representing the claims.
+   */
+  public static JsonObject getClaims(String jwt) {
+    String encodedJson = jwt.split("\\.")[1];
+    String decodedJson = new String(Base64.getDecoder().decode(encodedJson));
+    return new JsonObject(decodedJson);
+  }
+
+  /**
+   * Returns true if the token is encrypted, otherwise false.
+   * @param token A string version of the token to test.
+   * @return True if the token is encrypted, otherwise false.
+   * @throws TokenValidationException
+   */
+  public static boolean isEncrypted(String token) throws TokenValidationException {
+    // This is based on how JOSEObject parses encrypted tokens. It checks the number of "parts"
+    // which are equivalent to the number of "." separators in the token. If there are 5 parts
+    // it is considered to be encrypted. If there are not 5 it is rejected from JWE parsing.
+    // Our unencrypted tokens have 3 parts (and 2 separators).
+    int parts = StringUtils.countMatches(token, ".");
```

Comment:
No need to add the complete commons-lang3 library - you can easily make a for loop to count the dots (optionally: put the loop into a `countDots(String) method).
Since version 3.4 `StringUtils.countMatches(token, ".")` can be replaced by `StringUtils.countMatches(token, '.')`.

Reply:
Agree. Why not `token.split("\\.").length`? For some reason I was thinking about counting a character...

Reply:
`token.split("\\.").length` is a nice hack - it creates strings and then throws them away. The final result is correct, the code is somewhat misleading.

```
    long dots = token.chars().filter(c -> c == '.').count();
```